### ISSUE TITLE
no recipes for auto sieve

### ISSUE THREAD

**Issue by adamcirillo** [Quantum64/ExGregilo#2]
using version 1.0.3 and have no recipes for the gregtech auto sieve

**Comment by Quantum64** [comment]
Alright I'll look into it.

**Comment by Quantum64** [comment]
I need your GT5U version and log output please.

**Comment by adamcirillo** [comment]
Gtu 5.0.9.26 using ex gregilo 1.0.3 
[fml-client-latest.txt](https://github.com/Quantum64/ExGregilo/files/638125/fml-client-latest.txt)

**Comment by ghost** [comment]
The recipes work if you use per-item settings in MaterialProperties.cfg

(it enables the full material list, and then they work)

Tested that this morning, with 1.0.3 and .09.26

**Comment by Quantum64** [comment]
So for some reason the latest GT5U broke GT_ModHandler.addCraftingRecipe, and I'm still waiting from a response from Ash.  Anyways hopefully I'll have some more time to work on this over the holidays, as I'll have some free time.  I hope to add ways to obtain the gems through sifting.

**Comment by adamcirillo** [comment]
I changed in the gregtech.cfg 
B:EnableAllComponents=true
B:EnableAllMaterials=true

recipes show up now for some reason might be the stainless steel spring maybe? I think that's a high requirement for a lv/mv machine

**Comment by ghost** [comment]
Here are my recipes, which I find to be more fair/balanced/sensible, if you wish to use them.
They use GTs own tier-specific materials (determined by main material in Tier machine hulls, e.g. steel aluminium etc)

![image](https://cloud.githubusercontent.com/assets/2331387/21014147/d0c15188-bd2a-11e6-8594-d0a7b2fbdc1d.png)

```
P  Me P
A  M  C
P  H  P

P = Plate, Me = Mesh, A = Robot Arm, M = Motor, C = Conveyor, H = Hull

LV  = P: Steel,           Me: Steel,           A: LV,  M: LV,  C:LV,  H: LV
MV  = P: Aluminium,       Me: Aluminium,       A: MV,  M: MV,  C:MV,  H: MV
HV  = P: Stainless Steel, Me: Stainless Steel, A: HV,  M: HV,  C:HV,  H: HV
EV  = P: Titanium,        Me: Titanium,        A: EV,  M: EV,  C:EV,  H: EV
IV  = P: TungstenSteel,   Me: TungstenSteel,   A: IV,  M: IV,  C:IV,  H: IV
LuV = P: Chrome,          Me: Chrome,          A: LuV, M: LuV, C:LuV, H: LuV
ZPM = P: Iridium,         Me: Iridium,         A: ZPM, M: ZPM, C:ZPM, H: ZPM
UV  = P: Osmium,          Me: Osmium,          A: UV,  M: UV,  C:UV,  H: UV
```

**Comment by Quantum64** [comment]
I would love to use them as soon as I can get Asp to tell me why the GregTech recipe added isn't working.  I could probably use the GameRegistry and materials from GT_OreDictUnificator but that wasn't and shouldn't be the recommended way to add machine recipes.

**Comment by ghost** [comment]
It's working if the material system is enabled (by default all unused materials are disabled)
To speed up GT postinit, they restructured the material enum and material registration in 5.09.24

From what I recall, GT's recipe manager now scans the recipe list, and disables all materials not found in all registered recipes. (also scans Minetweaker scripts, and enables those, now too)
So assuming you register the recipes properly using the proper methods, GT will scan them, enable the required pieces, and go from there.

It's dropped nearly 50-60% of load time 👍 

I'd get ahold of BloodAsp and see if he can help you work things out.

**Comment by ghost** [comment]
Also, another note.

I created custom mesh recipes (most are disabled and return null recipes)
I used material appropriate rods instead of wires (since not all materials have wires in the material registry)

So, I think if you follow that suit, and use my machine recipes, also change all mesh recipes to use 8x rod (of mesh material) + 1x small gear (of mesh material)
e.g. 8x Steel Rod + 1x Small Steel Gear, etc 👍 

```
R R R
R G R
R R R

R = Rod, G = Gear
```

Also, some of the mesh recipes (Naquadah, Osmium, from top of my head) have invalid NBT values set for the Clicks Required, and result in erratic sifting behavior (material disappears from hand, ItemStack renders mid-air, moving up 1/8th of block per click)

**Comment by Quantum64** [comment]
Again, I'm totally open to a PR if you want to fix that before I'm able to: https://github.com/Quantum64/ExGregilo/blob/master/src/main/java/co/q64/exgregilo/links/gregtech/tools/WireMesh.java#L29

I can't seem to reproduce the issue with invalid NBT on meshes, it seems to work fine when they're spawned in anyway.  The only issue I can think of that might caused undefined behavior is if this value https://github.com/Quantum64/ExGregilo/blob/master/src/main/java/co/q64/exgregilo/tile/AdvancedSieveTile.java#L115 is negative, but I don't see how that could be possible.

**Comment by Quantum64** [comment]
Fixed in 1203101fa5e690d3fba7f2db8521bd2e3bb85a9f
